File: php/problem_3.php
<?php
// What is the largest prime factor of the number 600851475143?
function largest_prime_factor($number = 600851475143, $prime = 2)
{
  while ($prime ** 2 <= $number) {
    ($number % $prime == 0) ? ($number /= $prime) : ($prime += ($prime > 2 ? 2 : 1));
  }

  return $number;
}

$anwser = 6857;
echo largest_prime_factor() == $anwser ? "true" : "false";

File: python/problem_3.py
def largest_prime_factor(number=600851475143, prime=2):
    while (prime ** 2 <= number):
        if (number % prime == 0):
            number //= prime
        else:
            prime += 2 if prime > 2 else 1
    return number


answer = 6857
print(True if largest_prime_factor() == answer else False)
